diff --git includes/handlers.inc includes/handlers.inc
index 4741793..d0e58b0 100644
--- includes/handlers.inc
+++ includes/handlers.inc
@@ -288,8 +288,9 @@ class views_handler extends views_object {
     $this->localization_keys = array(
       $display_id,
       $this->handler_type,
-      $options['table'],
-      $options['field']
+	  // NOTE: use (unique) id instead of field
+      // $options['field']
+  	  $options['id']
     );
 
     $this->unpack_options($this->options, $options);
@@ -314,6 +315,11 @@ class views_handler extends views_object {
       }
     }
 
+    // not sure if this is required
+    if (isset($options['id'])) {
+      $this->id = $options['id'];
+  	}
+
     $this->query = &$view->query;
   }
 
diff --git plugins/views_plugin_display.inc views/plugins/views_plugin_display.inc
index 65aaf0d..f18c2be 100644
--- plugins/views_plugin_display.inc
+++ plugins/views_plugin_display.inc
@@ -26,7 +26,7 @@
 class views_plugin_display extends views_plugin {
   /**
    * The top object of a view.
-   * 
+   *
    * @var view
    */
   var $view = NULL;
@@ -2607,9 +2607,10 @@ class views_plugin_display extends views_plugin {
         $handler_type = $type;
       }
       $handler = views_get_handler($info['table'], $info['field'], $handler_type);
+      // Note: use (unique) $info['id'] instead of $info['field']
       if ($handler) {
         $handler->init($this->view, $info);
-        $handler->unpack_translatables($translatable, array_merge($parents, array($handler_type, $info['table'], $info['field'])));
+        $handler->unpack_translatables($translatable, array_merge($parents, array($handler_type, $info['table'], $info['id'])));
       }
 
       // Prevent reference problems.
